packages/react-instantsearch/src/core/highlight.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+
+function extractTagName(highlightedValue) {
+  const tagParse = (/<(.+?)>/).exec(highlightedValue);
+  return tagParse === null ? null : tagParse[1];
+}
+
+/**
+ * This function transforms an highlighted response
+ * to an array of React elements. This should be
+ * used in place of dangerouslySetInnerHTML when
+ * using the highlightReasult provided by Algolia.
+ *
+ * The function autodetects which tag is used in
+ * results to highlight, and creates the corresponding
+ * React element.
+ *
+ * It also creates keys on React elements to avoid warnings.
+ *
+ * @param {string} attribute the name of the attribute for which we want the highlighted value
+ * @param {object} hit the hit from the algolia response
+ * @returns {Array.<string|ReactElement>} the array of react element
+ * @example
+ * const ProductHits = Hits.connect(({hits}) => {
+ *   const hitComponents = hits.map( hit =>
+ *     <div key={hit.objectID}>
+ *       <span className="hit-name">{hightlight('name', hit)}</span>
+ *     </div>
+ *   );
+ *
+ *   return <div className="hits">{hitComponents}</div>;
+ * });
+ */
+export default function hightlight(attribute, hit) {
+  if (!hit) throw new Error('The hit containing the attribute should be provided');
+  if (!{}.hasOwnProperty.call(hit, attribute)) throw new Error(`${attribute} should be a retrievable attribute.`);
+  if (!hit._highlightResult ||
+      !{}.hasOwnProperty.call(hit._highlightResult, attribute)){
+    throw new Error(`${attribute} should be an highlighted attribute`);
+  }
+
+  const highlightedValue = hit._highlightResult[attribute].value;
+  const tag = extractTagName(highlightedValue);
+  const firstPassSplit = highlightedValue.split(`<${tag}>`);
+  const firstValue = firstPassSplit.shift();
+  const elements = firstValue === '' ? [] : [firstValue];
+
+  firstPassSplit.forEach((split, i) => {
+    const s = split.split(`</${tag}>`);
+    elements.push(
+      React.createElement(
+        tag,
+        {
+          key: `split-${i}-${s[0]}`,
```

Comment:
Because React. Each time you use an array variable as a child of an Element, React will warn you about the keys. Then why? It's because the key identifies the element in the array (not it's position) which gives React the ability to change the DOM in an optimized way (move instead of destroy for example)